src/OrbitBase/Logging.cpp

```diff
@@ -5,15 +5,65 @@
 #include "OrbitBase/Logging.h"
 
 #include <array>
+#include <system_error>
 
 #include "absl/base/const_init.h"
 #include "absl/debugging/stacktrace.h"
 #include "absl/debugging/symbolize.h"
+#include "absl/strings/str_cat.h"
 #include "absl/strings/str_format.h"
 #include "absl/synchronization/mutex.h"
 
 static absl::Mutex log_file_mutex(absl::kConstInit);
 std::ofstream log_file;
+const int kLogFileLifetimeMinutes = 60 * 24 * 7;  // one week
+
+std::optional<int> GetLogFileLifetimeMinutes(std::string log_file_name) {
+  unsigned first_delimiter_pos = log_file_name.find_first_of(kLogFileNameDelimiter);
+  unsigned last_delimiter_pos = log_file_name.find_last_of(kLogFileNameDelimiter);
+  std::string tm_str =
+      log_file_name.substr(first_delimiter_pos + 1, last_delimiter_pos - first_delimiter_pos - 1);
+  if (tm_str.empty()) return std::nullopt;
+
+  std::istringstream istr(tm_str);
+  int year, month, day, hour, minute, second;
+  char delimiter;
+  // The timestamp in the log file name follows a format of "%Y_%m_%d_%H_%M_%S".
+  if (istr >> year >> delimiter >> month >> delimiter >> day >> delimiter >> hour >> delimiter >>
+      minute >> delimiter >> second) {
+    // std::tm will set the year on basis of 1900 and month on basis of January.
+    std::tm log_file_tm = {second, minute, hour, day, month - 1, year - 1900};
+    std::chrono::system_clock::time_point log_file_tp =
+        std::chrono::system_clock::from_time_t(std::mktime(&log_file_tm));
+    std::chrono::system_clock::time_point now = std::chrono::system_clock::now();
+    std::chrono::minutes log_file_lifetime_minutes =
+        std::chrono::duration_cast<std::chrono::minutes>(now - log_file_tp);
+    return log_file_lifetime_minutes.count();
+  } else {
+    return std::nullopt;
+  }
+}
+
+ErrorMessageOr<void> TryRemoveOldLogFiles(const std::filesystem::path& log_dir) {
+  std::string error_message;
+  for (const auto& log_file_path : std::filesystem::recursive_directory_iterator(log_dir)) {
+    std::string log_file_name = log_file_path.path().filename().u8string();
```

Comment:
does u8string yield result different from just string()?

Reply:
Ah, I used `u8string()` when debugging for some issues and forgot to change back. Sorry about that. `u8string()` makes no difference with `string()`. I changed to use `string()`.